aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -116,9 +119,9 @@ SparseTensor new_with_dims_and_tensor_sparse(
 /** Public creation API that dispatch to methods above **/
 
 /** Empty init **/
-Tensor empty_sparse(IntArrayRef size, const TensorOptions& options, c10::optional<MemoryFormat> optional_memory_format) {
-  TORCH_CHECK(!options.pinned_memory(), "Only dense CPU tensors can be pinned");
-  return new_with_dims_sparse(size.size(), 0, size, options);
+Tensor empty_sparse(IntArrayRef size, c10::optional<ScalarType> dtype, c10::optional<Layout> layout, c10::optional<Device> device, c10::optional<bool> pin_memory, c10::optional<MemoryFormat> optional_memory_format) {
+  TORCH_CHECK(!pin_memory.has_value() || !*pin_memory, "Only dense CPU tensors can be pinned");
```

Comment:
I saw this idiom in a few places and it was weird to me that you didn't just use the `blah_or_default` functions here. I guess it's something to do with asserts?

Reply:
I found this more expressive since it's a precondition check for the function and this way it explicitly states the allowed values for the argument. With `pinned_memory_or_default` you'd have to look up that function to know if the `nullopt` case is allowed or not.